src/rocketchat.coffee

```diff
@@ -153,12 +153,13 @@ class RocketChatBotAdapter extends Adapter
 
 					if curts > @lastts
 						@lastts = curts
+						
+						user = @robot.brain.userForId newmsg.u._id, name: newmsg.u.username
```

Comment:
In the old way, it's looking for a user with the given ID and username in the new room.  

Because the brain hasn't been updated yet it still has the user in the old room.  This mean that `brain.userForId()` returns a new user object and any extra data saved in the brain for that user is lost.

By dropping `room: newmsg.rid` and setting the room on the next line the user is found in the brain (so the extra data is not lost), and the room the user is in is still updated and persisted in the brain.